devicemodel/hw/pci/passthrough.c

```diff
@@ -626,7 +621,11 @@ cfginitbar(struct vmctx *ctx, struct passthru_dev *ptdev)
 		 */
 		if (bartype == PCIBAR_MEM64) {
 			i++;
-			assert(i <= PCI_BARMAX);
+			if (i > PCI_BARMAX) {
+				warnx("BAR count out of range\n");
+				return -1;
```

Comment:
So if this happens, we can gracefully handle it, right?

Reply:
Normally, it will not happen, If this happens, it would be a bug in firmware/bios (assuem SOS doesn't reprogram the BAR register), it should be fixed in firmware/bios. I think just return error code is fine. What do you think?